cub/cub/device/dispatch/dispatch_transform.cuh

```diff
@@ -0,0 +1,862 @@
+                   // larger). May not be aligned to alignof(T). E.g.: array of int3 starting at address 4, ptr == 0
+  int head_padding; // byte offset between ptr and the original pointer. Value inside [0;15] or [0;127].
+
+  _CCCL_HOST_DEVICE const T* ptr_to_elements() const
+  {
+    return reinterpret_cast<const T*>(ptr + head_padding);
+  }
+
+  _CCCL_HOST_DEVICE friend bool operator==(const aligned_base_ptr& a, const aligned_base_ptr& b)
+  {
+    return a.ptr == b.ptr && a.head_padding == b.head_padding;
+  }
+};
+
+template <typename T>
+_CCCL_HOST_DEVICE auto make_aligned_base_ptr(const T* ptr, int alignment) -> aligned_base_ptr<T>
+{
+  const char* base_ptr = round_down_ptr(ptr, alignment);
+  return aligned_base_ptr<T>{base_ptr, static_cast<int>(reinterpret_cast<const char*>(ptr) - base_ptr)};
+}
+
+constexpr int bulk_copy_alignment     = 128;
+constexpr int bulk_copy_size_multiple = 16;
+
+_CCCL_DEVICE _CCCL_FORCEINLINE bool select_one()
+{
+#if CUB_PTX_ARCH >= 900
+  const ::cuda::std::uint32_t membermask = ~0;
+  ::cuda::std::uint32_t is_elected;
+  asm volatile(
+    "{\n\t .reg .pred P_OUT; \n\t"
+    "elect.sync _|P_OUT, %1;\n\t"
+    "selp.b32 %0, 1, 0, P_OUT; \n"
+    "}"
+    : "=r"(is_elected)
+    : "r"(membermask)
+    :);
+  return threadIdx.x < 32 && static_cast<bool>(is_elected);
+#else // CUB_PTX_ARCH >= 900
+  return false;
+#endif // CUB_PTX_ARCH >= 900
+}
+
+#ifdef _CUB_HAS_TRANSFORM_UBLKCP
+// TODO(bgruber): inline this as lambda in C++14
+template <typename Offset, typename T>
+_CCCL_DEVICE void bulk_copy_tile(
+  ::cuda::std::uint64_t& bar,
+  int tile_stride,
+  char* smem,
+  int& smem_offset,
+  ::cuda::std::uint32_t& total_bytes_bulk_copied,
+  Offset global_offset,
+  const aligned_base_ptr<T>& aligned_ptr)
+{
+#  if CUB_PTX_ARCH >= 900
+  _CCCL_IF_CONSTEXPR (alignof(T) > bulk_copy_alignment)
+  {
+    if (smem_offset > 0)
+    {
+      smem_offset += int{alignof(T)};
+    }
+  }
+  const char* src = aligned_ptr.ptr + global_offset * sizeof(T);
+  char* dst       = smem + smem_offset;
+  assert(reinterpret_cast<uintptr_t>(src) % bulk_copy_alignment == 0);
+  assert(reinterpret_cast<uintptr_t>(dst) % bulk_copy_alignment == 0);
+
+  // TODO(bgruber): we could precompute bytes_to_copy on the host
+  const int bytes_to_copy = round_up_to_po2_multiple(
+    aligned_ptr.head_padding + static_cast<int>(sizeof(T)) * tile_stride, bulk_copy_size_multiple);
+
+  ::cuda::ptx::cp_async_bulk(::cuda::ptx::space_cluster, ::cuda::ptx::space_global, dst, src, bytes_to_copy, &bar);
+
+  total_bytes_bulk_copied += bytes_to_copy;
+
+  smem_offset += static_cast<int>(sizeof(T)) * tile_stride; // always divisible by bulk_copy_alignment and alignof(T)
+  _CCCL_IF_CONSTEXPR (alignof(T) <= bulk_copy_alignment)
+  {
+    smem_offset += bulk_copy_alignment; // Don't merge into the other alignment check at the beginning for better
+                                        // codegen. bgruber measured it!
```

Comment:
The `alignof(T)` could be different for the next tile, so the idea was to add `bulk_copy_alignment` in case the alignment was smaller/equal, and add the alignment instead if it was bigger. Of course I messed this up, because the if here should check for the alignment of the next tile's `T`.

Therefore, I will drop this handling.